pkg/config/configuration.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"gopkg.in/yaml.v3"
+)
+
+// File is an on-disk configuration of a Config.
+type Configuration struct {
+	Registry Registry `mapstructure:"registry"`
+	TokenCmd string   `mapstructure:"token_cmd" yaml:"token_cmd"` // run in shell to generate token
```

Comment:
I think it would be helpful to name this consistently with the command used to set it.. "generator" or "source"

Reply:
Seeking consistency with `gcloud`, how about `registry auth set-token-source` and `registry auth print-token`? 

Or could setting this be handed with `registry config set token-source`?

Reply:
Agreed. Once we decide on that.